File: bubble-sort/BubbleSort.js
function bubbleSort1(array) {
    let isSorted = false;
    while(!isSorted) {
        isSorted = true;
        let index = 1;
        while(index < array.length) {
            if(array[index - 1] > array[index]) {
                isSorted = false;
                [array[index - 1], array[index]] = [array[index], array[index - 1]];
            }
            index++;
        }
    }
    return array;
}

function bubbleSort2(array) {
    let isSorted = false;
    while(!isSorted) {
        isSorted = true;
        let index = 0;
        while(index < array.length - 1) {
            if(array[index] > array[index + 1]) {
                isSorted = false;
                [array[index + 1], array[index]] = [array[index], array[index + 1]];
            }
            index++;
        }
    }
    return array;
}

function bubbleSort3(array) {
    let isSorted = false;
    while(!isSorted) {
        isSorted = true;
        let index = array.length - 1;
        while(index > 0) {
            if(array[index] < array[index - 1]) {
                isSorted = false;
                [array[index], array[index - 1]] = [array[index - 1], array[index]]
            }
            index--;
        }
    }
    return array;
}

exports.bubbleSort = bubbleSort1;

File: bubble-sort/BubbleSort.java
public class BubbleSort {
    public static int[] bubbleSort1(int[] array) {
        boolean isSorted = false;
        while(!isSorted) {
            isSorted = true;
            int index = 1;
            while(index < array.length) {
                if(array[index - 1] > array[index]) {
                    isSorted = false;
                    int temp = array[index - 1];
                    array[index - 1] = array[index];
                    array[index] = temp;
                }
                index++;
            }
        }
        return array;
    }

    public static int[] bubbleSort2(int[] array) {
        boolean isSorted = false;
        while(!isSorted) {
            isSorted = true;
            int index = 0;
            while(index < array.length - 1) {
                if(array[index] > array[index + 1]) {
                    isSorted = false;
                    int temp = array[index + 1];
                    array[index + 1] = array[index];
                    array[index] = temp;
                }
                index++;
            }
        }
        return array;
    }

    public static int[] bubbleSort3(int[] array) {
        boolean isSorted = false;
        while(!isSorted) {
            isSorted = true;
            int index = array.length - 1;
            while(index > 0) {
                if(array[index] < array[index - 1]) {
                    isSorted = false;
                    int temp = array[index - 1];
                    array[index - 1] = array[index];
                    array[index] = temp;
                }
                index--;
            }
        }
        return array;
    }
}
